### PR TITLE
refactor: Use react-native-test-app to manage our test app

### PR DESCRIPTION
**Pull request by Saadnajmi**
# Overview

This PR replaces the test app in the repo with one managed by [React Native Test App](https://github.com/microsoft/react-native-test-app). This is mainly inspired by my want to add macOS support, which is currently blocked by / tracked at #141. Still, setting up the template test app now will make standing up the macOS portion of the test app trivial come time.

"React Native Test App" provides test apps for all platforms as a package. It handles the native bits, allowing you the developer to focus on the JS. There are a few advantages to using this package:

- It makes RN version updates easier, as it will handle the proper native code changes
- It makes targeting new platforms (like macOS or Windows) easier, since it has a template to configure each platform's test app.

## Steps 

I tried to document my steps well with my individual commit messages. In essence:
```
mv example/ example2 # Move the existing test app out of the way for a bit
yarn add -D react-native-test-app
yarn init-test-app
# The options I chose
# ✔ What is the name of your test app? … MenuExample
# ✔ Which platforms do you need test apps for? › Android, iOS
# ✔ Where should we create the new project? … example
cd example

```
After this I did a few extra things:
- Added generated lock files after running `yarn / pod install` in the new example test app folder
  - `cd example && yarn && pod install --project-directory:ios`
  - `git add yarn.lock ios/Podfile.lock`
- Ran `yarn lint --fix` to fix lint errors in the generated test app (mostly double quotes vs single quotes)
- Modified the example test app's `package.json` to include the following dependency so we can actually _test_ menu:
  - `"@react-native-menu/menu": "../"`

Then finally, I copied over the JS source from the old test app, and deleted the old test app
```
rm example/index.js example/App.js
mv example2/index.tsx example.index.tsx
mv example2/src/App.tsx example/src/App.tsx
rm -rf example2/
```

# Test Plan

Locally ran the iOS and Android test apps. 

Screen Shot 2021-06-27 at 4 16 11 PM

### CODE REVIEW

**Comment by Saadnajmi** [comment]
Since I couldn't add you as reviewers..
@tido64 who maintains react-native-test-app 
@Naturalclar who maintains react-native menu

Alas, I couldn't add macOS support because `react-native-macos` isn't on 0.64 yet :(. I decided to go forward with moving the test app anyway to make it easier to add macOS support in the fut
Keeping this as a draft at the moment because `run-ios` doesn't work. If that's not a blocker, I can make it a full PR. Working on filing the right bugs in the cli now =)

File: example/react-native.config.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            "android",
+            "app",
+            "src",
+            "main",
+            "AndroidManifest.xml"
+          )
+        ),
+      },
+      ios: {
+        project: "ios/ReactTestApp-Dummy.xcodeproj",
```
Comment (tido64) [inline example/react-native.config.js:49]: Have you tried setting this to:

```suggestion
      project: "node_modules/.generated/ios/ReactTestApp.xcodeproj",
```

`yarn ios` will output an error but start the app anyway:

```
% yarn ios
yarn run v1.22.10
$ react-native run-ios
error Could not find "Podfile.lock" at null.lock. Did you run "pod install" in iOS directory?
error Could not find the following native modules: ReactTestApp-DevSupport. Did you forget to run "pod install" ?
info Found Xcode workspace "MenuExample.xcworkspace"
info Launching iPhone 12 (iOS 14.5)
info Building (using "xcodebuild -workspace MenuExample.xcworkspace -configuration Debug -scheme MenuExample -destination id=55FCD6EA-B6C0-4491-ABE9-074D19045567")
success Successfully built the app
info Installing "~/Library/Developer/Xcode/DerivedData/MenuExample-gjjkxqorvpxgfpamfozsblsaxpie/Build/Products/Debug-iphonesimulator/ReactTestApp.app"
info Launching "com.microsoft.ReactTestApp"
success Successfully launched the app on the simulator
```

I will try to fix the error but it seems like this PR can proceed?

Comment (tido64) [reply example/react-native.config.js:49]: (Also very strange that it used to stop after the error when I first tried this a couple of weeks ago.)

Comment (tido64) [reply example/react-native.config.js:49]: Never mind. I did things in the wrong order.

Comment (tido64) [reply example/react-native.config.js:49]: Just an FYI: I have a fix for the CLI but it's taking a while to get reviewed/merged: https://github.com/react-native-community/cli/pull/1444

Comment (tido64) [reply example/react-native.config.js:49]: This should now be fixed in [@react-native-community/cli-platform-ios@5.0.2](https://www.npmjs.com/package/@react-native-community/cli-platform-ios/v/5.0.2).

File: example/yarn.lock [hunk 2]
```diff
@@ -987,6 +987,19 @@
     plist "^3.0.1"
     xcode "^2.0.0"
 
+"@react-native-community/cli-platform-ios@^5.0.2":
```
Comment (tido64) [inline example/yarn.lock:990]: You might want to dedupe this. We have two versions in here now. Or just regenerate `yarn.lock` from scratch.

File: example/package.json [hunk 3]
```diff
@@ -19,13 +19,14 @@
   "devDependencies": {
     "@babel/core": "^7.12.9",
     "@babel/runtime": "^7.12.5",
+    "@react-native-community/cli-platform-ios": "^5.0.2",
```
Comment (tido64) [inline example/package.json:22]: This shouldn't be necessary.

**Comment by Saadnajmi** [comment]
@Naturalclar This should be ready for review! It should also make updating to RN 0.65 very simple, I just didn't want to bundle that in the same review. Thanks @tido64 for helping fix upstream CLI issues!

**Review by Naturalclar — APPROVED** [review]
@Saadnajmi @tido64 awesome! thank you guys so much!

**Comment by tido64** [comment]
> Alas, I couldn't add macOS support because `react-native-macos` isn't on 0.64 yet :(.

Btw, the first 0.64 versions of `react-native-macos` is quite usable now ;)

**Comment by Saadnajmi** [comment]
> > Alas, I couldn't add macOS support because `react-native-macos` isn't on 0.64 yet :(.
> 
> Btw, the first 0.64 versions of `react-native-macos` is quite usable now ;)

Yep! I have a working menu component I wrote for macOS elsewhere that I can port here once I get the chance!

**Pull request merged**